helix-term/src/ui/document.rs

```diff
@@ -36,14 +37,23 @@ impl<F: FnMut(&mut TextRenderer, LinePos)> LineDecoration for F {
 /// that merges the layered highlights to create the final text style
 /// and yields the active text style and the char_idx where the active
 /// style will have to be recomputed.
-struct StyleIter<'a, H: Iterator<Item = HighlightEvent>> {
+///
+/// IS_BYTE_ITER indicates wether the source iterator provides byte or char
+/// ranges
+///
+/// TODO(ropey2): hopefully onueday helix and ropey will operate entierly
+/// on byte ranges and we can remove this
+struct StyleIter<'a, H: Iterator<Item = HighlightEvent>, const IS_BYTE_ITER: bool> {
```

Comment:
I did this because sadly enums ca not be used as const parameters yet and I wanted to avoid this check in the hot loop. But its probably a better idea to just let the compiler figure this. Enum is definitely more readable